packages/sdk-middleware-auth/src/base-auth-flow.js

```diff
@@ -92,7 +100,8 @@ function executeRequest(
       })
     })
     .catch((error: Error) => {
-      response.reject(error)
+      if (response && typeof response.reject === 'function')
```

Comment:
Seems like with node-fetch in some cases you actually can not and do not have to reject. At least that's what the tests say if you leave this check out.